### ISSUE TITLE
Pylint no-member warnings in 4.5.4.58

### ISSUE THREAD

**Issue by robertschweizer** [opencv/opencv-python#570]
### Expected behaviour

Pylint was previously able to inspect `cv2` member existence with the `extension-pkg-whitelist` parameter.

### Actual behaviour

For opencv-python-4.5.4.58, pylint can no longer inspect the members.

### Steps to reproduce

On Windows 10, Python 3.8 64bit, using Pylint 2.11.1:

```py
"""Test module test.py"""
from cv2 import getStructuringElement, MORPH_ELLIPSE

a = getStructuringElement(MORPH_ELLIPSE, (3, 3))
```

With opencv-python-4.5.3.56:

```
$ pylint test.py --extension-pkg-whitelist=cv2

-------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 5.00/10, +5.00)
```

With opencv-python-4.5.4.58:

```
$ pylint test.py --extension-pkg-whitelist=cv2
************* Module test
test.py:2:0: E0611: No name 'getStructuringElement' in module 'cv2' (no-name-in-module)
test.py:2:0: E0611: No name 'MORPH_ELLIPSE' in module 'cv2' (no-name-in-module)

----------------------------------------------------------------------
Your code has been rated at -40.00/10 (previous run: 10.00/10, -50.00)
```

##### Issue submission checklist

 - [x] This is not a generic OpenCV usage question (looking for help for coding, other usage questions, homework etc.)
   
 - [x] I have read the README of this repository and understand that this repository provides only an automated build toolchain for OpenCV Python packages (there is no actual OpenCV code here)
   
 - [x] The issue is related to the build scripts in this repository, to the pre-built binaries or is a feature request (such as "please enable this additional dependency")
 - [x] I'm using the latest version of ``opencv-python``

**Comment by robertschweizer** [comment]
Good idea, maybe the C extension name just changed. But no, with `--unsafe-load-any-extension=y` it also does not work.

**Comment by asmorkalov** [comment]
Since 4.5.4.60 pylint works with OpenCV python loader correctly. Command line example:
```
python3 -m pylint --extension-pkg-whitelist=cv2,numpy ./contours.py
```

**Comment by robertschweizer** [comment]
The fix for this was reverted in 5633cf319efadb628e8369b48834d74491c10b1a, so this bug appears again in version 4.6.0.66. Should we rename and reopen this issue or create a new one?

**Comment by pasbi** [comment]
I get the `Module 'cv2' has no '*' member`-`pylint`-error with
- :x: `python3-opencv` (4.6.0 installed with `apt`)
- :x: `opencv-python` (4.6.0 installed with `pip`)
- :x: self-built OpenCV (using GitHub repos `opencv/opencv` and `opencv/opencv_contrib` *but not using `opencv-python`*).
- :heavy_check_mark: The only way I found to make `pylint` work was to install version **4.5.5** of `opencv-python` with **pip**.

However, I'm required to build OpenCV from source for technical reasons.
What's the charm that lets the 4.5.5 pip wheel work with pylint and how can I apply it to my self-built OpenCV?

<sub>Build and test environment is a debian bookworm Docker container.</sub>
